### PR TITLE
docs(react-grid): update filtering guide

### PR DESCRIPTION
**Pull request by gsobolev**

### CODE REVIEW

File: packages/dx-react-grid/docs/guides/filtering.md [hunk 1]
```diff
@@ -11,33 +11,29 @@ The following plugins implement filtering features:
```
Comment (dirk-pieterse) [inline packages/dx-react-grid/docs/guides/filtering.md]: "The filtering state management, Filter Row rendering, and filtering logic are implemented in the following plugins:" and then
"The following plugins implement filtering features:"? Why do you repeat this?

Comment (gsobolev) [reply packages/dx-react-grid/docs/guides/filtering.md]: Looks like there is no need in the last sentence of the first paragraph. I'll remove it. Thanks!

File: packages/dx-react-demos/src/material-ui/filtering/custom-filter-row.jsx [hunk 2]
```diff
@@ -16,81 +13,73 @@ import {
...
-  filter: PropTypes.object,
 };
 
-SexFilterCellBase.defaultProps = {
+UnitsFilterCellBase.defaultProps = {
   filter: null,
 };
 
-const SexFilterCell = withStyles(styles, { name: 'SexFilterCell' })(SexFilterCellBase);
+const UnitsFilterCell = withStyles(styles, { name: 'SexFilterCell' })(UnitsFilterCellBase);
```
Comment (YashZolmajdi) [inline packages/dx-react-demos/src/material-ui/filtering/custom-filter-row.jsx]: missed `{ name: 'SexFilterCell' }`, Looks so out of place in official [demo documentation](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/)

Comment (SergeyAlexeev) [reply packages/dx-react-demos/src/material-ui/filtering/custom-filter-row.jsx]: @YashZolmajdi thank you for pointing that. The related PR #1750

**Pull request merged**